src/LaunchDarkly.EventSource/EventSourceService.cs

```diff
@@ -140,6 +106,36 @@ public EventSourceService(Configuration configuration)
 
         }
 
+        protected virtual IStreamReader GetStreamReader(Stream stream)
+        {
+            return new EventSourceStreamReader(stream);
+        }
+
+        protected virtual async Task ProcessResponseFromReaderAsync(Action<string> processResponse, IStreamReader reader,
+            CancellationToken cancellationToken)
+        {
+
+            // In a loop, reading from a stream. while reading from the stream, process the lines in the stream.
+            // Reset the timer after processing each line (reading from the stream).
+            // If the read time out occurs, throw exception and exit method.
+
+            while (!cancellationToken.IsCancellationRequested)
```

Comment:
Do we not need a check against `reader.EndOfStream` after all?

Reply:
While reviewing the .net framework code for the EndOfStream property, it turns out the this property actually tries to read from the internal stream. What this does when connected to the EventSource API is wait for a "read" (response) from the stream and never proceeds into the while loop. So it turns out this does cause an issue because the read timeout is inside the while loop (which never gets reached in the case where the "read" is waiting). 

I tried moving the read timer outside this while loop but that resulted in a different problem in that the timer wasn't reset after parsing each "read" in the stream. 

I believe the code is correct now. Read timeouts do occur when they should and the timer is reset after each read.